salt/modules/boto_kinesis.py

```diff
@@ -441,6 +441,28 @@ def reshard(stream_name, desired_size, force=False,
     return r
 
 
+def list_streams(region=None, key=None, keyid=None, profile=None):
+    '''
+    Return a list of all streams visible to the current account
+
+    CLI example::
+
+        salt myminion boto_kinesis.list_streams
+    '''
+    conn = _get_conn(region=region, key=key, keyid=keyid, profile=profile)
+    streams = []
+    ExclusiveStartStreamName = ''
+    while ExclusiveStartStreamName is not None:
+        args = {'ExclusiveStartStreamName': ExclusiveStartStreamName} if ExclusiveStartStreamName else {}
+        r = _execute_with_retries(conn, 'list_streams', **args)
+        if 'error' in r:
+            return []
```

Comment:
This doesn't seem to be a good way to do the error handling. It takes an error and makes it look like it's not an error. If the backend gives back an error, we should return an error, rather than an empty list.

Reply:
@tkwilliams ?

Reply:
Eh, I'm more of a `log.error('Hey, your shiz is broked') ; return []` kinda guy, but the model you're using in the module seems to be `return {'error': 'Hey, your shiz is broked'}` so I can definitely do that to maintain stylistic consistency in the module :)

Should be up in a few minutes...

Reply:
We return an error so that the state module can report the errors back up to the end-user in the state output. it also allows us to clearly determine if something was an error, or we really meant to return an empty list, dict, or a None value.